internal/database/teams_test.go

```diff
@@ -0,0 +1,431 @@
+		err := store.CreateTeam(ctx, team)
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestTeams_GetListCount(t *testing.T) {
+	internalCtx := actor.WithInternalActor(context.Background())
+	logger := logtest.NoOp(t)
+	db := NewDB(logger, dbtest.NewDB(logger, t))
+	johndoe, err := db.Users().Create(internalCtx, NewUser{Username: "johndoe"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	alice, err := db.Users().Create(internalCtx, NewUser{Username: "alice"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	store := db.Teams()
+
+	createTeam := func(team *types.Team, members ...int32) *types.Team {
+		team.CreatorID = johndoe.ID
+		if err := store.CreateTeam(internalCtx, team); err != nil {
+			t.Fatal(err)
+		}
+		for _, m := range members {
+			if err := store.CreateTeamMember(internalCtx, &types.TeamMember{TeamID: team.ID, UserID: m}); err != nil {
+				t.Fatal(err)
+			}
+		}
+		return team
+	}
+
+	engineeringTeam := createTeam(&types.Team{Name: "engineering"}, johndoe.ID)
+	salesTeam := createTeam(&types.Team{Name: "sales"})
+	supportTeam := createTeam(&types.Team{Name: "support"}, johndoe.ID)
+	ownTeam := createTeam(&types.Team{Name: "sgown", ParentTeamID: engineeringTeam.ID}, alice.ID)
+	batchesTeam := createTeam(&types.Team{Name: "batches", ParentTeamID: engineeringTeam.ID}, johndoe.ID, alice.ID)
+
+	t.Run("GetByID", func(t *testing.T) {
+		for _, want := range []*types.Team{engineeringTeam, salesTeam, supportTeam, ownTeam, batchesTeam} {
+			have, err := store.GetTeam(internalCtx, want.ID)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if diff := cmp.Diff(have, want); diff != "" {
+				t.Fatal(diff)
+			}
+		}
+		t.Run("not found error", func(t *testing.T) {
+			_, err := store.GetTeam(internalCtx, 100000)
+			if err == nil {
+				t.Fatal("no error for not found team")
+			}
+			tnfe := &TeamNotFoundError{}
+			if !errors.As(err, tnfe) {
+				t.Fatalf("invalid error returned, expected not found got %v", err)
+			}
+		})
+	})
+
+	t.Run("ListCountTeams", func(t *testing.T) {
+		allTeams := []*types.Team{engineeringTeam, salesTeam, supportTeam, ownTeam, batchesTeam}
+
+		// Get all.
+		haveTeams, haveCursor, err := store.ListTeams(internalCtx, ListTeamsOpts{})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if diff := cmp.Diff(haveTeams, allTeams); diff != "" {
+			t.Fatal(diff)
+		}
+
+		if haveCursor != 0 {
+			t.Fatal("incorrect cursor returned")
+		}
+
+		// Test cursor pagination.
+		var lastCursor int32
+		for i := 0; i < len(allTeams); i++ {
+			t.Run(fmt.Sprintf("List 1 %s", allTeams[i].Name), func(t *testing.T) {
+				opts := ListTeamsOpts{LimitOffset: &LimitOffset{Limit: 1}, Cursor: lastCursor}
+				teams, c, err := store.ListTeams(internalCtx, opts)
+				if err != nil {
+					t.Fatal(err)
+				}
+				lastCursor = c
+
+				if len(teams) != 1 {
+					t.Fatalf("expected exactly 1 team, got %d", len(teams))
+				}
+
+				if diff := cmp.Diff(teams[0], allTeams[i]); diff != "" {
+					t.Fatal(diff)
+				}
```

Comment:
```suggestion
				if diff := cmp.Diff(teams, allTeams[i:i+1]); diff != "" {
					t.Fatal(diff)
				}
```

nit: with `cmp.Diff` I'd just go and compare the slices rather than checking the length and then comparing the elements. In spite of the message being pretty vocal here in case there is length difference - we do not which elements we've got. `cmp.Diff` gives us more information, and the code is shorter.

Reply:
(I'd say similar throughout the test file)